src/rp2040.ts

```diff
@@ -1640,12 +1649,14 @@ export class RP2040 {
       const Rd = opcode & 0x7;
       const leftValue = this.registers[Rn];
       const rightValue = this.registers[Rm];
-      const result = (leftValue - rightValue) | 0;
+      const result = leftValue - rightValue;
       this.registers[Rd] = result;
-      this.N = (leftValue | 0) < (rightValue | 0);
-      this.Z = leftValue === rightValue;
+      this.N = !!(result & 0x80000000);
+      this.Z = (result & 0xffffffff) === 0;
```

Comment:
Just for your information - `result | 0` would also do the same (shrink the value to 32 bits). But it's also fine this way!